app/components/windows/ChildWindow.vue.ts

```diff
@@ -64,10 +59,15 @@ export default class ChildWindow extends Vue {
       return;
     }
 
-    this.$set(this.components, this.activeComponentInd, {
-      name: this.options.componentName,
-      isShown: this.options.isShown
-    });
+    // This is essentially a race condition, but make a best effort
+    // at having a successful paint cycle before loading a component
+    // that will do a bunch of synchronous IO.
+    setTimeout(() => {
+      this.$set(this.components, this.activeComponentInd, {
+        name: this.options.componentName,
+        isShown: this.options.isShown
+      });
+    }, 50);
```

Comment:
would `requestAnimationFrame` work better here? i think that waits like exactly one paint cycle, but i might be wrong about that

Reply:
You're totally right, it does get called on every paint cycle.  However, it gets called right before the paint cycle as opposed to after.  I still think maybe there's something I could with it, but I'm not really sure.  Electron also has a paint event I'd like to play around with a bit more.  However I don't really have time to mess around with it more at the moment.

Reply:
Haha interesting.  Theoretically that should work I think?  I'll test some more when I have time.

Reply:
`Vue.$nextTick()` ?

Reply:
@holiber `Vue.$nextTick()` unfortunately does not work.  `$nextTick()` only guarantees that the changes have been made in the DOM, but it doesn't guarantee that the browser has actually painted the changes to the screen.  My experimentation using `$nextTick()` often resulted in us blocking the pain to the screen with synchronous IO.